modules/heat_conduction/src/constraints/GapConductanceConstraint.C

```diff
@@ -39,9 +56,23 @@ GapConductanceConstraint::computeQpResidual(Moose::MortarType mortar_type)
       return -_lambda[_qp] * _test_secondary[_i][_qp];
     case Moose::MortarType::Lower:
     {
-      auto l = (_phys_points_primary[_qp] - _phys_points_secondary[_qp]).norm();
-      return (_k * (_u_primary[_qp] - _u_secondary[_qp]) / l - _lambda[_qp]) * _test[_i][_qp];
+      // we are creating an AD version of phys points primary and secondary here...
+      ADRealVectorValue ad_phys_points_primary = _phys_points_primary[_qp];
+      ADRealVectorValue ad_phys_points_secondary = _phys_points_secondary[_qp];
+
+      // ...which uses the derivative vector of the primary and secondary displacements as
+      // an approximation of the true phys points derivatives
+      for (unsigned int i = 0; i < _n_disp; ++i)
+      {
+        ad_phys_points_primary(i).derivatives() = (*_disp_primary[i])[_qp].derivatives();
+        ad_phys_points_secondary(i).derivatives() = (*_disp_secondary[i])[_qp].derivatives();
+      }
+
+      auto l =
+          std::max((ad_phys_points_primary - ad_phys_points_secondary) * _normals[_qp], _min_gap);
+      return (_lambda[_qp] - _k * (_u_primary[_qp] - _u_secondary[_qp]) / l) * _test[_i][_qp];
```

Comment:
And high k...probably need to test with k = 1 and k = 100? Or maybe it won't matter?

Reply:
I mean, push back if you want, I just know that cases like that in which are quick for you to test now can really head off confusion down the line

Reply:
That'll definitely make the scaling worse and the problem harder to solve...which seems to be what you want haha

Reply:
I don't think it'll ever go above 100, usually about 60. May go less than 1 when doing a gas gap, I forget exactly.